### PR TITLE
Transition startup entry on update

### PR DESCRIPTION
**Pull request by MDrakos**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-1435" title="DX-1435" target="_blank">StoryDX-1435</a>  The state-svc autostarts as an .app on macOS
  </td></table>
  <br />

### CODE REVIEW

File: internal/osutils/autostart/autostart_darwin.go [hunk 1]
```diff
@@ -20,6 +20,11 @@ const (
 )
 
 func enable(exec string, opts Options) error {
+	err := cleanLegacyFile(exec, opts)
+	if err != nil {
+		return errs.Wrap(err, "Could not clean legacy file")
+	}
```
Comment (Naatan) [inline internal/osutils/autostart/autostart_darwin.go:26]: We just started moving this logic to the installer, might be worth waiting for David's PR and piggy backing off his changes:

https://github.com/ActiveState/cli/pull/2456/files/f7027b520621686fcac30ec61731c2c03c58955b..479edd891a2492b2f2cd7d9af55e1b8839a96e92

File: internal/osutils/autostart/legacy_darwin.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	if err != nil {
+		return false, errs.Wrap(err, "Could not open plist file")
+	}
+
+	decoder := plist.NewDecoder(reader)
+	var p LegacyPlist
+	err = decoder.Decode(&p)
+	if err != nil {
+		return false, errs.Wrap(err, "Could not decode plist file")
+	}
```
Comment (Naatan) [inline internal/osutils/autostart/legacy_darwin.go:30]: Wouldn't these error conditions mean it's not a plist file? Are we not causing errors here for users who aren't upgrading from this legacy state?

**Review by Naatan — CHANGES_REQUESTED** [review]
In addition to the lines tagged; why is this PR showing so many vendor changes?

File: internal/osutils/autostart/legacy_darwin.go (hunk 2 above)
Comment (MDrakos) [reply internal/osutils/autostart/legacy_darwin.go:30]: Without knowing more about plist files I don't think we can make that assumption. 

For users upgrading who aren't in a legacy state the return on line 32 should be false. We still have the `ProgramArguments` key in the file, which is an array but there should only be one entry in the non-legacy case.

**Comment by MDrakos** [comment]
> In addition to the lines tagged; why is this PR showing so many vendor changes?

I added the library `howett.net/plist` to work with plist files then ran `go mod tidy` and `go mod vendor`

**Review by Naatan — CHANGES_REQUESTED** [review]
This one still needs to be addressed:

https://github.com/ActiveState/cli/pull/2475/files#r1161891070

Recognize you may be blocked for now, but I'm not the blocker so bouncing it back ;)

**Comment by MDrakos** [comment]
> This one still needs to be addressed:
> 
> https://github.com/ActiveState/cli/pull/2475/files#r1161891070
> 
> Recognize you may be blocked for now, but I'm not the blocker so bouncing it back ;)

Whoops looks like the PR is merged, so I was able to piggyback my changes with the new `Update` function. Tested out on my local machine and it's working as expected.

**Review by Naatan — APPROVED** [review]

**Pull request merged**